### ISSUE TITLE
Issues with URL Path matching logic

### ISSUE THREAD

**Issue by UltraNemesis** [igm/sockjs-go#81]
Because of the way URL path matching is done in the library using Regular Expressions and MatchString against the HTTP request, the overall path matching in this library is defective.

For example, if I register a handler with the prefix "echo", all of below match and work even though only the first one should be working. As long at the last part of the URL ends with the prefix string, the library will allow that URL to be handled. 

http://server:port/echo/
http://server:port/test/echo/
http://server:port/echo/test/test/echo/

**Comment by igm** [comment]
entire URL matching implementation would need a rewrite. There have been many mux solutions created since this was created and probably we should use one of the existing ones (e.g. https://github.com/gorilla/mux, https://github.com/gin-gonic/gin ,...) Not sure which one to go with, probably something lightweight would be a preference. I'm open for advices

**Comment by FZambia** [comment]
I suppose adding `^` to the beginning of all match strings should fix most of these situations? I mean that we can extend recent pr bc95a2e3f22d4c973820a5cb7179354a8d86fd29 to include `^` not only in empty prefix case but always. I was indended to do this first but then decided to do an accurate change to not break sth.

**Comment by igm** [comment]
Originally when I implemented handlers roughy 5 years ago the flexibility there was implemented on purpose, just to validate last parts of the URL path, so all the cases @UltraNemesis mentions are not a failure but rather that was implemented like that on purpose. But I understand it is confusing behaviour.

Today, I'd omit `prefix` completely and leave the handling of prefix on the user of the library, i.e. using `http.StripPrefix(...)` or similar mechanism.  

I don't know if there are any users that depend on "confusing" behaviour of the last part and I'm not sure how many would be affected by adding `^` to the beginning of all match strings even though I completely agree to do it. So I would not add this into `v2`.

We can restrict path matching with `^` on master and in following weeks release it as `v3`. There are multiple improvements in `master`, but also breaking changes compared to `v2` and probably it would make sense to release `v3` soon (i.e. to create a `go.mod` with `module github.com/igm/sockjs-go/sockjs/v3` and tag `v3.0.0`) so that `go mod` projects can use version `v3`.
